python/lsst/ts/m2gui/display/gauge.py

```diff
@@ -0,0 +1,166 @@
+# This file is part of ts_m2gui.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["Gauge"]
+
+from PySide2.QtCore import Qt, QSize
+from PySide2.QtGui import QPainter, QColor
+from PySide2.QtWidgets import QWidget
+
+
+class Gauge(QWidget):
+    """Draws gauge with the color scale.
+
+    Parameters
+    ----------
+    magnitude_min : `float` or `int`
+        Minimal magnitude in gauge.
+    magnitude_max : `float` or `int`
+        Maximal magnitude in gauge.
+    """
+
+    # Minimum size of widget
+    MIN_SIZE = 100
+
+    def __init__(self, magnitude_min, magnitude_max):
+        super().__init__()
+
+        self._min = 0
+        self._max = 0
+        self._update_range(magnitude_min, magnitude_max)
+
+        self.setMinimumSize(self.MIN_SIZE, self.MIN_SIZE)
+        self.setMaximumWidth(2 * self.MIN_SIZE)
+
+    def _update_range(self, magnitude_min, magnitude_max):
```

Comment:
You don't need that. You can merge _update_range with set_magnitude_range (see above for why the world magnitude is bad choice), and leave that in only one function

Reply:
Fixed in the commit: [af8c4ea](https://github.com/lsst-ts/ts_m2gui/pull/13/commits/af8c4eada9478d63f692c5cd50c53c057e3e2b34). Thanks!